src/tools/perf/cuda/cuda_alloc.c

```diff
@@ -69,15 +71,31 @@ uct_perf_cuda_alloc_reg_mem(const ucx_perf_context_t *perf,
                             unsigned flags,
                             uct_allocated_memory_t *alloc_mem)
 {
+    uct_md_attr_v2_t md_attr = {.field_mask = UCT_MD_ATTR_FIELD_REG_ALIGNMENT};
+    size_t reg_align;
+    void *reg_address;
+    size_t reg_length;
     ucs_status_t status;
 
+    status = uct_md_query_v2(perf->uct.md, &md_attr);
+    if (status != UCS_OK) {
+        ucs_error("uct_md_query_v2() returned %d", status);
+        return status;
+    }
+
     status = ucx_perf_cuda_alloc(length, mem_type, &alloc_mem->address);
     if (status != UCS_OK) {
         return status;
     }
 
-    status = uct_md_mem_reg(perf->uct.md, alloc_mem->address,
-                            length, flags, &alloc_mem->memh);
+    /* Register memory respecting MD reg_alignment */
+    reg_align   = ucs_max(md_attr.reg_alignment, 1);
```

Comment:
why needed? is there an MD that returns 0 here? if yes it should be fixed

Reply:
Well, as I see it's 0s for all MDs, expect gdr_copy:
```
[     INFO ] >>> knem 0
[     INFO ] >>> cma 0
[     INFO ] >>> posix 0
[     INFO ] >>> sysv 0
[     INFO ] >>> xpmem 0
[     INFO ] >>> cuda_cpy 0
[     INFO ] >>> cuda_ipc 0
[     INFO ] >>> mlx5_0 0
[     INFO ] >>> mlx5_1 0
[     INFO ] >>> mlx5_2 0
[     INFO ] >>> gdr_copy 65536
```
And the second reason: I looked at the existing handling of this param in production code: https://github.com/openucx/ucx/blob/master/src/ucp/core/ucp_mm.c#L525
```
        if (context->rcache == NULL) {
            reg_align = ucs_max(context->tl_mds[md_index].attr.reg_alignment, 1);
            ucs_align_ptr_range(&reg_address, &reg_length, reg_align);
        }
```
Should we set this attribute to 1 everywhere by default?
I quickly checked places where this field is used, should be ok to change it to 1

Reply:
> Should we set this attribute to 1 everywhere by default?
Yes